packages/h5wasm/src/local/worker.utils.ts

```diff
@@ -0,0 +1,149 @@
+import type {
+  Attribute,
+  ChildEntity,
+  Group,
+  ProvidedEntity,
+} from '@h5web/shared/hdf5-models';
+import { EntityKind } from '@h5web/shared/hdf5-models';
+import { buildEntityPath, getNameFromPath } from '@h5web/shared/hdf5-utils';
+import { ready as h5wasmReady, ready } from 'h5wasm';
+
+import { parseDType } from '../utils';
+
+export async function initH5Wasm(): typeof ready {
+  const module = await ready;
+
+  // Throw HDF5 errors instead of just logging them
+  module.activate_throwing_error_handler();
+
+  return module;
+}
+
+export async function mountWorkerFS(mountPt: string) {
+  const { FS } = await h5wasmReady;
+  const { WORKERFS } = FS.filesystems;
+
+  FS.mkdir(mountPt);
+  return FS.mount(WORKERFS as FS.FileSystemType, {}, mountPt);
+}
+
+export function parseEntity(
+  h5wasm: Awaited<typeof h5wasmReady>,
+  fileId: bigint,
+  path: string,
+  isChild: true,
+): ChildEntity;
+
+export function parseEntity(
+  h5wasm: Awaited<typeof h5wasmReady>,
+  fileId: bigint,
+  path: string,
+  isChild?: false,
+): ProvidedEntity;
+
+export function parseEntity(
```

Comment:
This essentially duplicates the `parseEntity` utility in [`h5wasm/src/utils`](https://github.com/silx-kit/h5web/blob/main/packages/h5wasm/src/utils.ts#L86) but takes the `h5wasm` module and a `fileId` as params instead of an H5Wasm `entity` object. Nothing prevents the current provider from adopting the same approach as the new provider, so refactoring is definitely on the agenda ... but no rush.